index.js

```diff
@@ -192,8 +199,12 @@ function Jasmine2HTMLReporter(options) {
     function loadFlakedSuiteNames() {
         var storedSuiteNames = storage.getItem(storageUID) || {};
 
-        if (typeof storedSuiteNames === 'string') {
-            return JSON.parse(storedSuiteNames);
+        if (_.isString(storedSuiteNames)) {
+            try {
+                return JSON.parse(storedSuiteNames);
+            } catch (error) {
+                log('Error retrieving flaked suite names', error);
```

Comment:
Do you want to still be returning storedSuiteNames if this exception occurs? Maybe null or an empty object would be more appropriate?